Client/Frontend/Browser/Tab.swift

```diff
@@ -679,3 +682,66 @@ class TabWebViewMenuHelper: UIView {
         }
     }
 }
+
+// MARK: - Brave Search
+
+extension Tab {
+    /// Call the api on the Brave Search website and passes the fallback results to it.
+    /// Important: This method is also called when there is no fallback results
+    /// or when the fallback call should not happen at all.
+    /// The website expects the iOS device to always call this method(blocks on it).
+    func injectResults() {
+        let dispatchGroup = DispatchGroup()
+        var shouldInjectResults = true
+        
+        dispatchGroup.enter()
+        DispatchQueue.main.async {
+            // If the backup search results happen before the Brave Search loads
+            // The method we pass data to is undefined.
+            // For such case we do not call that method or remove the search backup manager.
+            // swiftlint:disable:next safe_javascript
+            self.webView?.evaluateJavaScript("window.onFetchedBackupResults === undefined") {
+                result, error in
+                defer { dispatchGroup.leave() }
+                
+                if let error = error {
+                    log.error("onFetchedBackupResults existence check error: \(error)")
+                }
+                
+                guard let resultBool = result as? Bool else {
+                    log.error("onFetchedBackupResults existence check, failed to unwrap bool result value")
+                    return
+                }
+                
+                if resultBool {
+                    shouldInjectResults = false
+                }
+            }
+        }
+        
+        dispatchGroup.notify(queue: .main) { [weak self] in
+            guard let self = self else { return }
+            if !shouldInjectResults {
+                log.info("Search Backup results are ready but the page has not been loaded yet")
+                return
+            }
+                
+            var queryResult = "null"
+            
+            if let url = self.webView?.url,
+               BraveSearchManager.isValidURL(url),
+               let result = self.braveSearchManager?.fallbackQueryResult {
+                queryResult = result
+            }
+            
+            self.webView?.evaluateSafeJavaScript(
```

Comment:
does this get called regardless of whether the webview URL is search.brave.com? as mentioned above, `shouldInjectResults` can be true on any website.

Reply:
It only gets called for brave-search domains, you can look for `func isValidURL` in code, same applies to previous comment.
I can add second check in this method to be double sure

Reply:
@iccub i see https://github.com/brave/brave-ios/pull/3748/files#diff-0245cd17e421c32b1b43a7650b4915db51305407ba64872399960c24d1218d1aR722 but this code is outside that block, so i think it gets called regardless? (but queryResult will be empty)

Reply:
that's correct, we check for the url already, i didn't see that

and we call that method regardless because the website does not know whether we are still fetching results or not, it actually blocks showing results until this call is returned